### PR TITLE
Avoid double rebuild.

### LINKED ISSUE #1030: Todo example is not efficient when click on an item
**Describe what scenario you think is uncovered by the existing examples/articles**
When clicking on an item, that TodoItem widget is rebuilt 2 times in a row

**Describe why existing examples/articles do not cover this case**
Line 263 of main.dart:
```
child: ListTile(
  onTap: () {
    itemFocusNode.requestFocus();        //this makes the widget rebuilt 1st time
    textFieldFocusNode.requestFocus();   //this makes the widget rebuilt 2nd time because it also triggers the onFocusChange of the Focus widget
  },
```

### PR DESCRIPTION
**Pull request by ronnieeeeee**
Hi Remi, fixes #1030.

https://github.com/rrousselGit/river_pod/pull/1042#issuecomment-1013690851
If it is better to implement useIsfocused as suggested by Remi here, I would appreciate a hint. 
I will implement it.
I'm not good at English, so sorry if it's hard to understand.

### CODE REVIEW

File: examples/todos/lib/main.dart [hunk 1]
```diff
@@ -261,19 +257,15 @@ class TodoItem extends HookConsumerWidget {
           }
         },
         child: ListTile(
-          onTap: () {
-            itemFocusNode.requestFocus();
-            textFieldFocusNode.requestFocus();
-          },
+          onTap: () => isFocused.value = true,
```
Comment (rrousselGit) [inline examples/todos/lib/main.dart:260]: Focus is the responsibility of FocusNode

Not using FocusNode isn't reasonable, as it'll break the logic of "focusing one item unfocus the previously focused item"

Comment (ronnieeeeee) [reply examples/todos/lib/main.dart:260]: Thank you for the advice, I learned a lot.
I'm going to try to fix it.

Comment (rrousselGit) [reply examples/todos/lib/main.dart:260]: Making a useIsFocused should be the solution.

Comment (rrousselGit) [reply examples/todos/lib/main.dart:260]: Yes this is fine

I didn't mean to add a new hook to flutter_hooks, just making one for the example.

Comment (ronnieeeeee) [reply examples/todos/lib/main.dart:260]: @davidmartos96 Oh, I learned a lot from that!

Comment (ronnieeeeee) [reply examples/todos/lib/main.dart:260]: @davidmartos96 
Hi David, Do you plan to submit the code you fixed for additional PR?
If David doesn't plan to submit a PR, I'll add it to this PR.

Comment (ronnieeeeee) [reply examples/todos/lib/main.dart:260]: Thanks, David, I will add that 👍

"I will add it to this PR." might have been a rude English expression.
To be precise, I wanted to tell David "I would like to add to this PR" or "I want to add to this PR."
My English is not very good, I'm still learning.

Comment (davidmartos96) [reply examples/todos/lib/main.dart:260]: > Thanks, David, I will add that 👍
> 
> "I will add it to this PR." might have been a rude English expression.
> To be precise, I wanted to tell David "I would like to add to this PR" or "I want to add to this PR."
> My English is not very good, I'm still learning.

No worries 😉

Comment (ronnieeeeee) [reply examples/todos/lib/main.dart]: Hi, Remi,
Updated code to include useIsFocused.
Would you take a look at the updated code?

File: examples/todos/lib/main.dart [hunk 2]
```diff
@@ -282,3 +280,18 @@ class TodoItem extends HookConsumerWidget {
...
+bool useIsFocused(FocusNode node) {
+  final hasFocusVN = useState(node.hasFocus);
+
+  useEffect(() {
+    void l() {
+      hasFocusVN.value = node.hasFocus;
+    }
+
+    node.addListener(l);
+    return () => node.removeListener(l);
```
Comment (rrousselGit) [inline examples/todos/lib/main.dart:293]:
```suggestion
    void listener() {
      hasFocusVN.value = node.hasFocus;
    }

    node.addListener(listener);
    return () => node.removeListener(listener);
```

File: examples/todos/lib/main.dart [hunk 3]
```diff
@@ -282,3 +280,18 @@ class TodoItem extends HookConsumerWidget {
...
+  }, [node]);
+
+  return hasFocusVN.value;
```
Comment (rrousselGit) [inline examples/todos/lib/main.dart:296]:
```suggestion
  final isFocused = useState(node.hasFocus);

  useEffect(() {
    void l() {
      isFocused.value = node.hasFocus;
    }

    node.addListener(l);
    return () => node.removeListener(l);
  }, [node]);

  return isFocused.value;
```

File: examples/todos/lib/main.dart [hunk 4]
```diff
@@ -282,3 +280,18 @@ class TodoItem extends HookConsumerWidget {
...
+  useEffect(() {
+    void listener() {
+      hasFocusVN.value = node.hasFocus;
+    }
+
+    node.addListener(listener);
+    return () => node.removeListener(listener);
+  }, [node]);
+
+  return hasFocusVN.value;
```
Comment (rrousselGit) [inline examples/todos/lib/main.dart:296]:
```suggestion
  final isFocused = useState(node.hasFocus);

  useEffect(() {
    void listener() {
      isFocused.value = node.hasFocus;
    }

    node.addListener(listener);
    return () => node.removeListener(listener);
  }, [node]);

  return isFocused.value;
```

**Comment by rrousselGit** [comment]
Thanks!

**Pull request merged**